### PR TITLE
Fix checkbashisms download ans SCA violations

### PR DESCRIPTION
**Pull request by SpacePossum**

### CODE REVIEW

**Comment by keradus** [comment]
sorry, `Update devscript location` vs content of PR is confusion. what location we gonna change + reasoning, can you share?

**Comment by jrmajor** [comment]
@keradus The changed variable `VERSION_CB` is used in `wget -qO- "https://deb.debian.org/debian/pool/main/d/devscripts/devscripts_${VERSION_CB}.tar.xz"`. For the old value, it returns 404, so it looks like the location from which the script is downloaded will be changed.

**Comment by SpacePossum** [comment]
all PR's are currently failing SCA set, for example:
https://github.com/FriendsOfPHP/PHP-CS-Fixer/runs/5232717511?check_suite_focus=true

this is because the location:

>https://deb.debian.org/debian/pool/main/d/devscripts/devscripts_2.21.7.tar.xz

returns 404

the updated location

>https://deb.debian.org/debian/pool/main/d/devscripts/devscripts_2.21.7~bpo11+1.tar.xz

works

However, since the install script of the SCA exits on the first error it hasn't run all the other tooling, like stan and psalm, either, which is what is the rest of the PR will be about.

File: src/Doctrine/Annotation/Tokens.php [hunk 1]
```diff
@@ -240,23 +240,21 @@ public function insertAt(int $index, Token $token): void
...
     }
 
-    /**
-     * {@inheritdoc}
-     *
-     * @throws \InvalidArgumentException
-     */
     public function offsetSet($index, $token): void
     {
+        // @phpstan-ignore-next-line as we type checking here
```
Comment (jrmajor) [inline src/Doctrine/Annotation/Tokens.php:245]:
```suggestion
        /** @var mixed $token */
```
Also, there's `treatPhpDocTypesAsCertain` option in PHPStan configuration, which would prevent it from reporting runtime type checks for variables that are typed in phpdocs, although given that the entire project is statically analyzed, I think turning it off would do more harm than good.

Comment (SpacePossum) [reply src/Doctrine/Annotation/Tokens.php:245]: Switching it off on raises a 3 items on this PR, maybe it is worth doing that? Not sure....

Comment (jrmajor) [reply src/Doctrine/Annotation/Tokens.php:245]: I don't think so. In most methods we trust PHPDoc types. If there are only a few places where we want to validate them at runtime, it's safer to add `@var mixed`, as that is how we treat this variable in this case. Switching `treatPhpDocTypesAsCertain` off would prevent PHPStan from reporting truly unnecessary checks that are there by mistake.

**Comment by SpacePossum** [comment]
lets get the PR's green first with this one, than we can refine in others if really needed

**Pull request closed without merging**